doc/kernel/threads/scheduling.rst

```diff
@@ -175,6 +175,39 @@ becomes the current thread, its non-preemptible status is maintained.
     Locking out the scheduler is a more efficient way for a preemptible thread
     to inhibit preemption than changing its priority level to a negative value.
 
+.. _metairq_priorities:
+
+Meta-IRQ Priorities
+===================
+
+When enabled (see :option:`CONFIG_NUM_METAIRQ_PRIORITIES`), there is a special
+subclass of cooperative priorities at the highest (numerically lowest)
+end of the priority space: meta-IRQ threads.  These are scheduled
+according to their normal priority, but also have the special ability
+to preempt all other threads (and other meta-irq threads) at lower
+priorities, even if those threads are cooperative and/or have taken a
+scheduler lock.
+
+This behavior makes the act of unblocking a meta-IRQ thread (by any
+means, e.g. creating it, calling k_sem_give(), etc.) into the
+equivalent of a synchronous system call when done by a lower
+priority thread, or an ARM-like "pended IRQ" when done from true
+interrupt context.  The intent is that this feature will be used to
+implement interrupt "bottom half" processing and/or "tasklet" features
```

Comment:
Ah, if the quoted items are jargon the reader might want to look up, it would be helpful to put in some link references, for example for the "bottom half" and "tasklet" items, you could add a reference to something such as this:

    The intent is that this feature will be be used to
    implement interrupt bottom-half processing and/or tasklet features
    in driver subsystems.  (See, for example, `Linux Device Drivers
    <https://www.safaribooksonline.com/library/view/linux-device-drivers/0596000081/ch09s05.html>`_
     for an explanation.)

Reply:
Can we do that later as a documentation addition?   I don't have the links handy, and honestly this usage isn't particularly weird in my eyes.